### ISSUE TITLE
Down

### ISSUE THREAD

**Issue by domenkozar** [madjar/howoldis#3]
http://howoldis.herokuapp.com/

cc @madjar

**Comment by madjar** [comment]
Mmmm, looks like it crashed, but the logs were empty, and restarting it solved the problem.

I don't like that kind of problem. It works now, though. Hopefully, it will continue working.

**Comment by domenkozar** [comment]
Again down

**Comment by madjar** [comment]
Mmmm, I fear it might be linked to the new heroku model that doesn't allow free dynos to run 24/24. I'll remove my pinging bot, and see if that stops breaking.

**Comment by domenkozar** [comment]
did you restart it? since it's still down

**Comment by madjar** [comment]
I did restart it, and it was up at some point.

I'll have to take a deep look at the logs, and find a way to find out why the process is crashing.

**Comment by domenkozar** [comment]
Its probably some date parsing

**Comment by madjar** [comment]
The logs look like this:

```
Sep 06 11:19:14 howoldis heroku/api: Scale to web=1 by georges.dubus@gmail.com 
Sep 06 11:19:48 howoldis heroku/web.1: Starting process with command `cabal run` 
Sep 06 11:19:55 howoldis app/web.1: Preprocessing executable 'howoldis' for howoldis-0.1.0.0... 
Sep 06 11:20:00 howoldis app/web.1: Linking dist/build/howoldis/howoldis ... 
Sep 06 11:20:04 howoldis heroku/web.1: State changed from starting to up 
Sep 06 11:20:08 howoldis heroku/router: at=info method=GET path="/" host=howoldis.herokuapp.com request_id=3e1227b2-759d-4229-ba40-65ef04166da7 fwd="MYHOMEIP" dyno=web.1 connect=0ms service=73ms status=200 bytes=2826 
Sep 06 11:54:06 howoldis heroku/web.1: State changed from up to down 
Sep 06 11:54:06 howoldis heroku/web.1: Idling 
Sep 06 11:54:09 howoldis heroku/web.1: Stopping all processes with SIGTERM 
Sep 06 11:54:11 howoldis heroku/web.1: Process exited with status 143 
Sep 06 15:12:07 howoldis heroku/router: at=error code=H10 desc="App crashed" method=GET path="/" host=howoldis.herokuapp.com request_id=0a76cf18-9df3-4874-a381-73fb76142083 fwd="MYHOMEIP" dyno= connect= service= status=503 bytes= 
```

It's strange, the way I read it it receive a SIGTERM when the dyno idles, and crashes as a result. I'll make it behave gracefully on sigterm and see if that fixes it.

**Comment by madjar** [comment]
I switched to stack, since that's what I used these days, and I changed the buildpack to https://github.com/mfine/heroku-buildpack-stack.

I didn't think that would change anything to the problem, but the app seems to handle restarting after idling. I'll leave it like that to check if that actually works.

If that doesn't, I've implemented SIGTERM handling: https://github.com/madjar/howoldis/tree/handle-sigterm
